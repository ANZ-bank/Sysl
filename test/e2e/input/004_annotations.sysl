ThatApp:
	foo:
		...
	bar:
		...
MyApp:
    ep1:
        ThatApp <- foo

    ep2:
        ThatApp <- foo

    ep3:
        ThatApp <- bar

MyApp:
    * :
        ThatApp <- foo [~abc]
