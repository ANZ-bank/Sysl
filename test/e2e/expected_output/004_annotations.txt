apps {
  key: "MyApp"
  value {
    name {
      part: "MyApp"
    }
    endpoints {
      key: "* "
      value {
        name: "* "
        stmt {
          call {
            target {
              part: "ThatApp"
            }
            endpoint: "foo"
          }
          attrs {
            key: "patterns"
            value {
              a {
                elt {
                  s: "abc"
                }
              }
            }
          }
        }
      }
    }
    endpoints {
      key: "ep1"
      value {
        name: "ep1"
        stmt {
          call {
            target {
              part: "ThatApp"
            }
            endpoint: "foo"
          }
          attrs {
            key: "patterns"
            value {
              a {
                elt {
                  s: "abc"
                }
              }
            }
          }
        }
      }
    }
    endpoints {
      key: "ep2"
      value {
        name: "ep2"
        stmt {
          call {
            target {
              part: "ThatApp"
            }
            endpoint: "foo"
          }
          attrs {
            key: "patterns"
            value {
              a {
                elt {
                  s: "abc"
                }
              }
            }
          }
        }
      }
    }
    endpoints {
      key: "ep3"
      value {
        name: "ep3"
        stmt {
          call {
            target {
              part: "ThatApp"
            }
            endpoint: "bar"
          }
        }
      }
    }
  }
}
apps {
  key: "ThatApp"
  value {
    name {
      part: "ThatApp"
    }
    endpoints {
      key: "bar"
      value {
        name: "bar"
        stmt {
          action {
            action: "..."
          }
        }
      }
    }
    endpoints {
      key: "foo"
      value {
        name: "foo"
        stmt {
          action {
            action: "..."
          }
        }
      }
    }
  }
}
